src/hints.rs

```diff
@@ -138,6 +141,7 @@ impl<'a> From<&'a NotificationHint> for MessageItem {
             &NotificationHint::Y(value)                 => (Y              .to_owned(), MessageItem::Int32(value)),
             &NotificationHint::Urgency(value)           => (URGENCY        .to_owned(), MessageItem::Byte(value as u8)),
             &NotificationHint::Custom(ref key, ref val) => (key            .to_owned(), MessageItem::Str(val.to_owned ())),
+            &NotificationHint::CustomInt(ref key, val)  => (key            .to_owned(), MessageItem::Int32(val)),
```

Comment:
that's what I was thinking, I then went on to do `Custom(String, MessageItem)` but `MessageItem` doesn't implement `Hash` or `Eq` so that was a no-go. Seems like maybe we want a second enum to list a bunch of types.